tensorflow/lite/micro/examples/mnist_lstm/train.py

```diff
@@ -0,0 +1,166 @@
+      patience=1)  #early stop if validation loss does not drop anymore
+  model.fit(x_train,
+            y_train,
+            epochs=epochs,
+            validation_split=0.2,
+            batch_size=32,
+            callbacks=[callback])
+  return model
+
+
+def save_tf_model(model, save_path):
+  """Save the trained LSTM model in tensorflow format
+
+    Args:
+        model (tf.keras.Model): the trained LSTM Model
+        save_path (str): path to save the model
+    """
+
+  batch_size, steps, input_size = 1, 28, 28
+  run_model = tf.function(lambda x: model(x))
+  concrete_func = run_model.get_concrete_function(
+      tf.TensorSpec([batch_size, steps, input_size], model.inputs[0].dtype))
+  model.save(save_path, save_format="tf", signatures=concrete_func)
+  print(f'TF model saved to {save_path}')
+
+
+def save_tflite_model(model, saved_path, optimize):
+  """Convert the saved TF model to tflite model, then save it as .tflite flatbuffer format
+
+    Args:
+        model (tf.keras.Model): the trained LSTM Model
+        save_path (str): path to save the model
+        optimize (bool): enable model conversion optimization (dynamic range quantization)
+    """
+  fixed_input = tf.keras.layers.Input(shape=[28, 28],
+                                      batch_size=1,
+                                      dtype=model.inputs[0].dtype,
+                                      name='fixed_input')
+  fixed_output = model(fixed_input)
+  run_model = tf.keras.models.Model(
+      fixed_input, fixed_output
+  )  #converter requires fixed shape input to work, alternative: b/225231544
+  converter = tf.lite.TFLiteConverter.from_keras_model(run_model)
+  save_name = 'lstm.tflite'
+  if optimize:
+    converter.optimizations = [tf.lite.Optimize.DEFAULT]
+    save_name = 'quantized_' + save_name
+    print('Model conversion with dynamic range quantization')
+  tflite_model = converter.convert()
+
+  if not os.path.exists(saved_path):
+    os.makedirs(saved_path)
+  with open(saved_path + '/' + save_name, 'wb') as f:
+    f.write(tflite_model)
+  print(f"Tflite model saved to {saved_path}")
+
+
+def main(epochs, save_path, save_raw_model, optimize_conversion):
+  """train and save LSTM model using keras
+
+    Args:
+        save_path (string): save path for the trained model
+    """
+  trained_model = train_lstm_model(epochs)
+  if save_raw_model:
+    save_tf_model(trained_model, save_path)
+  save_tflite_model(trained_model, save_path, optimize_conversion)
+
+
+if __name__ == '__main__':
+  parser = argparse.ArgumentParser(
+      description='Train and convert a LSTM model')
+
+  parser.add_argument('--epochs',
+                      type=int,
+                      default=1,
+                      help='number of epochs to train the model')
+  parser.add_argument('--save_path',
+                      metavar='p',
+                      default='/tmp/trained_model',
+                      help='the path to save the trained model')
```

Comment:
Maybe call this "directory" instead of "path" (as well as save_dir instead of save_path) as it's a bit misleading now.

Reply:
Got it.